### ISSUE TITLE
[ChoiceValidator] Can't validate against an empty choices array

### ISSUE THREAD

**Issue by fancyweb** [symfony/symfony#13853]
I know it's a special case but I think it should be working.

```php
 $form->add('name', 'choice', array(
    'expanded' => true,
    'multiple' => true,
    'choices' => array(
        'choice #1',
        'choice #2',
        'choice #3'
    ),
    'constraints' => new Choice(array(
        'choices' =>  array(),
        'multiple' => true
    ))
));
```

The field should be valid if none of the checkboxes are checked but a `ConstraintDefinitionException` is thrown ((https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Validator/Constraints/ChoiceValidator.php#L40) because an empty array is considered as `false` if the comparison is not strict.

I think we just have to add another test in the condition that throws this exception like `&& !is_array($constraint->choices)`.

**Comment by ogizanagi** [comment]
Providing an array of choices with a only `null` seems to do the job, but it seems more like a hack than a solution: 

```php
$form->add('name', 'choice', array(
    'expanded' => true,
    'multiple' => true,
    'choices' => array(
        'choice #1',
        'choice #2',
        'choice #3'
    ),
    'constraints' => new Choice(array(
        'choices' =>  array(null),
        'multiple' => true
    ))
));
```

**Comment by fancyweb** [comment]
It's not just for example purpose.

In my case the `choices` constraint option is an array variable that can be empty.
My `choice` form represents a question where you need to check valid answers from a list of choices, but all proposed answers are wrong. The `choices` constraint option represents valid answers. Because there isn't any valid answers, the array is empty... I also use the `min` and `max` constraint options to be sure all answers have been checked.

At the moment, your solution is working, but as you said it's a hack.

**Comment by ogizanagi** [comment]
@fancyweb : could you have a look to #13856 ?

**Comment by webmozart** [comment]
This is fixed in #14448.

**Comment by SoboLAN** [comment]
If it was fixed, can someone close this ? :)
